src/veriflow/flows.py

```diff
@@ -133,16 +132,17 @@ def fit(
             for idx in range(0, N, batch_size):
                 end = min(idx + batch_size, N)
                 try:
-                    sample = torch.Tensor(data_train_shuffle[idx:end]).to(device)
+                    sample = torch.Tensor(data_train_shuffle[idx:end])  
```

Comment:
casting to Tensor unneccessary, if data is already a tensor
```suggestion
                    sample = data_train_shuffle[idx:end]
                    if not isinstance(sample, torch.Tensor):
                        sample = torch.Tensor(sample)  
```